orc/src/main/java/org/apache/iceberg/orc/ORCSchemaUtil.java

```diff
@@ -377,15 +377,15 @@ private static boolean isSameType(TypeDescription orcType, Type icebergType) {
     }
   }
 
-  private static Optional<Integer> icebergID(TypeDescription orcType) {
+  static Optional<Integer> icebergID(TypeDescription orcType) {
     return Optional.ofNullable(orcType.getAttributeValue(ICEBERG_ID_ATTRIBUTE))
         .map(Integer::parseInt);
   }
 
   static int fieldId(TypeDescription orcType) {
-    String idStr = orcType.getAttributeValue(ICEBERG_ID_ATTRIBUTE);
-    Preconditions.checkNotNull(idStr, "Missing expected '%s' property", ICEBERG_ID_ATTRIBUTE);
-    return Integer.parseInt(idStr);
+    return icebergID(orcType)
+        .orElseThrow(() -> new NullPointerException(
+            String.format("Missing expected '%s' property", ICEBERG_ID_ATTRIBUTE)));
```

Comment:
What's the value of this change? It seems like unnecessary code churn to me since the behavior is the same. It also causes an `Optional` to be instantiated and discarded. Not a huge problem, but it doesn't support the idea of changing this.